cmd/provider/daemon.go

```diff
@@ -169,31 +169,31 @@ func daemonCommand(cctx *cli.Context) error {
 
 	log.Infow("Shutting down daemon")
 
-	ctx, cancel := context.WithTimeout(context.Background(), shutdownTimeout)
+	shutdownCtx, cancel := context.WithTimeout(context.Background(), shutdownTimeout)
```

Comment:
maybe consider `cctx.Context`?

Reply:
I specifically did not use `cctx.Context` because that is canceled when the daemon receives CTRL-C, and is used to _trigger_ shutdown.  The `shutdownCtx` is specifically there to give a deadline to complete a graceful shutdown, and if it times out then a hard shutdown follows.